Add documentation build system, restructure repository, and update docs

jungshadow opened the pull request:
Adds Sphinx to aid in building documentation into various formats

jungshadow commented on BUILD.md:
```diff
@@ -0,0 +1,48 @@
+- [Pipenv](https://pipenv.pypa.io/en/latest/)
+- pandoc 1.16
+- [Imagemagick](https://imagemagick.org/index.php)
+- [Inkscape](https://inkscape.org/)
```
Added Inkscape as a requirement. In the end, we'll only need Imagemagick or Inkscape, but not both. Just depends whether it's easier/better to convert the SVGs in Sphinx or pandoc. Since HTML can display SVGs without issue, I'd rather do it in pandoc.

JDziurlaj replied:
I had issues with Inkscape. It seems to be omitting the text in the SVGs. Can you validate?

jungshadow commented on WorkingModel/Activities/Architecture_Review_Methodology.md:
```diff
@@ -27,7 +27,7 @@ For each security service, the Architecture Review will identify the product com
 ## Workflow
 
-![Image of Architecture Review Methodology](Architecture_Review_Methodology_files/_19_0_3_43701b0_1585746146678_950815_14100.svg)
+![Image of Architecture Review Methodology](./Architecture_Review_Methodology_files/workflow.svg)
```
@JDziurlaj I'm not sure if there was a reason for the filename, but I changed it.

jungshadow commented on WorkingModel/README.md:
```diff
@@ -67,7 +67,7 @@ RABET-V evaluates the security, reliability, and usability of products and provi
 The RABET-V Process consist of eight activities, beginning with Product Submission. The process flow will vary for each iteration.  
 
-![RABET-V Process Model](https://github.com/it-dept-cis/RABET-V-Pilot/blob/master/WorkingModel/election_verification_proc_model-2020-0115.png "RABET-V Process Model")
+![RABET-V Process Model](./election_verification_proc_model-2020-0115.png "RABET-V Process Model")
```
Switched to a relative path since it was breaking the build.

jungshadow commented:
I'm tagging you to test this, @JDziurlaj, since I made some changes that might break your build. You'll need to put Inkscape and Imagemagick on your syspath. That should get around the need to hard code the executable in `conf.py`.

jungshadow commented:
Ok @JDziurlaj, @wilsoaa, @planglois925...I'm going to need a sanity check here before any of this gets merged in. I updated all the docs and restructured the repository with an eye toward keeping the program docs by themselves (i.e. anything that was ancillary to building the program doc was moved into a different location). I'm going to break this down a few ways to make it digestible.

Now, the docs live in `docs` (earth-shattering...I know). The make files are at the root-level of `docs` and the actual markdown docs are in `source`. I exploded some of the existing directories out to make them more reflective of the structure in the doc.

Here's how the `source` part of the repository looks now:

```sh
source/
├── Activities
│   ├── Architecture_Review_Methodology.md
│   ├── Decision_Trees
│   │   └── decisiontree_master.json
│   ├── Process_Assessment_Methodology.md
│   ├── Product_Verification_Process.md
│   ├── Provider_Submission.md
│   ├── README.md
│   ├── Reporting_Process.md
│   ├── Security_Claims_Validation.md
│   ├── Submission_Review_Process.md
│   ├── Testing_Rules_Determination.md
│   └── media
│       ├── election_verification_proc_model-2020-0115.png
│       └── workflow.svg
├── Appendices
│   ├── Documentation_Summary.md
│   └── RABET-V_Glossary.md
├── Overview
│   ├── Maturity_Indexes.md
│   └── Security_Services.md
├── README.md
├── Security_Services_Architectural_Maturity_Index
│   ├── README.md
│   ├── Security_Service_Construction_Maturity.md
│   └── Security_Service_Usage_Maturity.md
├── Security_Services_Capability_Maturity_Index
│   ├── Authentication_Capability_Maturity.md
│   ├── Authorization_Capability_Maturity.md
│   ├── Boundary_Protection_Capability_Maturity.md
│   ├── Data_Confidentiality_Capability_Maturity.md
│   ├── Data_Integrity_Capability_Maturity.md
│   ├── Injection_Prevention_Capability_Maturity.md
│   ├── Logging_Alerting_Capability_Maturity.md
│   ├── README.md
│   ├── Secret_Management_Capability_Maturity.md
│   ├── System_Integrity_Capability_Maturity.md
│   └── User_Session_Management_Capability_Maturity.md
├── Software_Development_Maturity_Index
│   ├── Accessibility_Maturity.md
│   ├── README.md
│   ├── Usability_Maturity.md
│   └── media
│       └── samm_v2_overview.png
├── conf.py
└── index.rst
```
The reStructuredText index file builds the docs in a way that follows the updated doc.

```rst
.. toctree::
   :maxdepth: 2
   :glob:

   README.md
   Overview/Security_Services.md
   Overview/Maturity_Indexes.md
   Activities/README.md
   Activities/Provider_Submission.md
   Activities/Submission_Review_Process.md
   Activities/Process_Assessment_Methodology.md
   Activities/Architecture_Review_Methodology.md
   Activities/Security_Claims_Validation.md
   Activities/Testing_Rules_Determination.md
   Activities/Product_Verification_Process.md
   Activities/Reporting_Process.md
   Security_Services_Capability_Maturity_Index/README.md
   Security_Services_Capability_Maturity_Index/Authentication_Capability_Maturity.md
   Security_Services_Capability_Maturity_Index/Authorization_Capability_Maturity.md
   Security_Services_Capability_Maturity_Index/Boundary_Protection_Capability_Maturity.md
   Security_Services_Capability_Maturity_Index/Data_Confidentiality_Capability_Maturity.md
   Security_Services_Capability_Maturity_Index/Data_Integrity_Capability_Maturity.md
   Security_Services_Capability_Maturity_Index/Injection_Prevention_Capability_Maturity.md
   Security_Services_Capability_Maturity_Index/Logging_Alerting_Capability_Maturity.md
   Security_Services_Capability_Maturity_Index/Secret_Management_Capability_Maturity.md
   Security_Services_Capability_Maturity_Index/System_Integrity_Capability_Maturity.md
   Security_Services_Capability_Maturity_Index/User_Session_Management_Capability_Maturity.md
   Security_Services_Architectural_Maturity_Index/README.md
   Security_Services_Architectural_Maturity_Index/Security_Service_Construction_Maturity.md
   Security_Services_Architectural_Maturity_Index/Security_Service_Usage_Maturity.md
   Software_Development_Maturity_Index/README.md
   Software_Development_Maturity_Index/Accessibility_Maturity.md
   Software_Development_Maturity_Index/Usability_Maturity.md
   Appendices/Documentation_Summary.md
   Appendices/RABET-V_Glossary.md
```

At the root of the repository, there are a few more changes. Everything is effectively loaded into the following folders:

- `diagrams`: Anything that's a general diagram and _potentially_ being generated by another program;
- `research`: Holds the economic model and the research plan (NB: I was teetering on putting the economic plan in the root and ultimately put it in research, but happy to revisit that decision);
- `resources`: Related docs and presentations
- `for_consideration`: Anything I'm still on the fence about

Here's the directory tree with the breakdown (NB: In the directory tree below, I removed the docs folder breakdown for brevity's sake):

```sh
rabet-v-pilot/.
├── BUILD.md
├── Pipfile
├── Pipfile.lock
├── README.md
├── STYLEGUIDE.md
├── diagrams
│   └── Architecture_Review_Methodology
│       ├── icon_class_1862347028.svg
│       ├── icon_class_1891408970.svg
│       ├── icon_dataobject_538267658.svg
│       ├── icon_dataobject_759932913.svg
│       └── workflow.svg
├── docs
├── for_consideration
│   ├── Participants
│   │   ├── Participant\ Invite
│   │   └── README.md
│   ├── Samples
│   │   ├── Architecture\ Diagrams
│   │   │   └── Foo\ ENR\ Architecture\ 02182020.pdf
│   │   └── Labeling
│   │       ├── README.md
│   │       └── images
│   │           ├── decompose_high.svg
│   │           ├── decompose_least.svg
│   │           ├── decompose_mid.svg
│   │           ├── decompose_mid_high.svg
│   │           ├── operation_criticality.svg
│   │           ├── ports.svg
│   │           ├── security_service.svg
│   │           ├── sensitive_epb.svg
│   │           ├── sensitive_epb_class.svg
│   │           ├── sensitive_err.svg
│   │           ├── subcomponents.svg
│   │           └── third_party.svg
│   └── Security_Service_Requirements
│       ├── Authentication_Requirements.md
│       ├── Authorization_Requirements.md
│       ├── Boundary_Protection_Requirements.md
│       ├── Data_Confidentiality_Requirements.md
│       ├── Data_Integrity_Requirements.md
│       ├── Injection_Prevention_Requirements.md
│       ├── Logging_Alerting_Requirements.md
│       ├── Secret_Management_Requirements.md
│       ├── System_Integrity_Requirements.md
│       └── User_Session_Management_Requirements.md
├── research
│   ├── Economic_Model
│   │   └── README.md
│   └── Research_Plan
│       ├── README.md
│       └── Research_Questions.md
└── resources
    ├── Elections_Tech-Ver-White_Paper-2020-0121.pdf
    ├── Presentations
    │   ├── RABET-V\ Program\ Description\ April.pptx
    │   ├── RABET-V\ Webinar-20200430\ 1639-1.mp4
    │   └── RABET-V\ Webinar-20200430\ 1639-1.vtt
    └── RABET-V\ Program\ Description\ 0.1\ Committee\ Review.docx
```

A few outstanding issues (but nothing that a couple of `git mv`s can't fix):

1. A few of these files should probably be renamed to match the title of the file (e.g. `Security_Service_Usage_Maturity.md` is titled **Security Service Use** in the document.
1. The `for_consideration` is honesty in advertising. I'm looking for thoughts about where these folders should be moved. They seem like they're in active development and maybe they should eventually be moved back into the docs folder. Just want to make sure.
1. Not sure where the `Samples` folder should go. It seems like the root-level would be best, but they should potentially be named something a bit more clear.

JDziurlaj commented:
Hi @jungshadow, thanks for this! A few suggestions:

- Remove `diagrams/Architecture_Review_Method` if they are no longer used. 
- Move `Decision_trees` to `for_consideration`
- Verify the build files are in the correct directory

jungshadow commented:
> Remove `diagrams/Architecture_Review_Method` if they are no longer used.

@JDziurlaj The Architectural Review workflow is still used—the smaller SVGs are currently ommited—and I copied that SVG over to the specific media directory. I'm wondering if we should store some Architectural diagram files outside the docs for working purposes (e.g. if you have a MagicDraw file that has the workflow, it could go here). I'm still happy to remove if you think we don't need it, though.

> Move Decision_trees to for_consideration

Will do.

> Verify the build files are in the correct directory

Yup. We're good!
